code/game/objects/structures/tank_dispenser.dm

```diff
@@ -88,13 +88,13 @@
 	switch(action)
 		if("plasma")
 			var/obj/item/weapon/tank/internals/plasma/tank = locate() in src
-			if(tank)
+			if(tank && Adjacent(usr))
 				usr.put_in_hands(tank)
 				plasmatanks--

```

Comment:
`else if(tank)
(indent)tank.move(src.get_turf())`

Indented to the same level as the if()